sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -585,34 +589,6 @@ private[hive] class HiveMetastoreCatalog(hive: HiveContext) extends Catalog with
         }
 
       case p: LogicalPlan if p.resolved => p
-
-      case p @ CreateTableAsSelect(db, tableName, child, allowExisting, None) =>
```

Comment:
OK, I see your point, but it seems more reasonable to keep both `CreateTableAsSelect` pattern matching, probably we can remove the duplicated code for the first matching block, what do you think? 
Hmm, and `hasStorageSpec` seems not right to me in previous implementation